.github/actions/tool-setup/action.yml

```
@@ -74,7 +74,7 @@ runs:
       uses: actions/cache@v2
       with:
         path: ${{ steps.composer-cache.outputs.dir }}
-        key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+        key: ${{ runner.os }}-composer-${{ hashFiles('$GITHUB_WORKSPACE/**/composer.lock') }}
```

I was able to resolve the issue with failing `hashFiles` call by using absolute path, as suggested in https://github.com/actions/cache/issues/753#issuecomment-1063229812

I'd like to note, that seems like a bug in one of the `actions/cache` dependencies, that should be fixed  with https://github.com/actions/toolkit/pull/1015

---

Hm, it did not help much. Seems like `hashFiles('$GITHUB_WORKSPACE/**/composer.lock')` returns an empty string: https://github.com/Automattic/jetpack/runs/5623206329?check_suite_focus=true#step:28:145

---

That's probably because it's looking for the literal path `$GITHUB_WORKSPACE/**/composer.lock` rather than substituting the variable.

Going by their docs, it looks like this might work.
```suggestion
        key: ${{ runner.os }}-composer-${{ hashFiles( format( '{0}/**/composer.lock', github.workspace ) ) }}
```
Although I don't think that will actually help. I think the real problem is in the fact that `tools/docker/` has unreadable subdirectories, as Adi worked around for other E2E tests in #23453.

---

I applied Adi's fix, and now it works as expected